code/modules/mob/living/simple_animal/hostile/space_dragon.dm

```diff
@@ -0,0 +1,727 @@
+	var/chosen_color
+
+/mob/living/simple_animal/hostile/space_dragon/Initialize(mapload)
+	. = ..()
+	gust = new
+	gust.Grant(src)
+	rift = new
+	rift.Grant(src)
+	small_sprite = new
+	small_sprite.Grant(src)
+	RegisterSignal(small_sprite, COMSIG_ACTION_TRIGGER, .proc/add_dragon_overlay)
+
+/mob/living/simple_animal/hostile/space_dragon/Login()
+	. = ..()
+	if(!chosen_color)
+		dragon_name()
+		color_selection()
+
+/mob/living/simple_animal/hostile/space_dragon/Life(delta_time = SSMOBS_DT, times_fired)
+	. = ..()
+	tiredness = max(tiredness - 1, 0)
+	for(var/mob/living/consumed_mob in src)
+		if(consumed_mob.stat == DEAD)
+			continue
+		playsound(src, 'sound/effects/splat.ogg', 50, TRUE)
+		visible_message("<span class='danger'>[src] vomits up [consumed_mob]!</span>")
+		consumed_mob.forceMove(loc)
+		consumed_mob.Paralyze(50)
+	if((rifts_charged == 3 || (SSshuttle.emergency.mode == SHUTTLE_DOCKED && rifts_charged > 0)) && !objective_complete)
+		victory()
+	if(riftTimer == -1)
+		return
+	riftTimer = min(riftTimer + 1, maxRiftTimer + 1)
+	if(riftTimer == (maxRiftTimer - 60))
+		to_chat(src, "<span class='boldwarning'>You have a minute left to summon the rift!  Get to it!</span>")
+		return
+	if(riftTimer >= maxRiftTimer)
+		to_chat(src, "<span class='boldwarning'>You've failed to summon the rift in a timely manner!  You're being pulled back from whence you came!</span>")
+		destroy_rifts()
+		empty_contents()
+		playsound(src, 'sound/magic/demon_dies.ogg', 100, TRUE)
+		QDEL_NULL(src)
+
+/mob/living/simple_animal/hostile/space_dragon/AttackingTarget()
+	if(using_special)
+		return
+	if(target == src)
+		to_chat(src, "<span class='warning'>You almost bite yourself, but then decide against it.</span>")
+		return
+	if(istype(target, /turf/closed/wall))
+		if(tearing_wall)
+			return
+		tearing_wall = TRUE
+		var/turf/closed/wall/thewall = target
+		to_chat(src, "<span class='warning'>You begin tearing through the wall...</span>")
+		playsound(src, 'sound/machines/airlock_alien_prying.ogg', 100, TRUE)
+		var/timetotear = 4 SECONDS
+		if(istype(target, /turf/closed/wall/r_wall))
+			timetotear = 12 SECONDS
+		tearing_wall = TRUE
+		if(do_after(src, timetotear, target = thewall))
+			if(istype(thewall, /turf/open))
+				return
+			thewall.dismantle_wall(1)
+			playsound(src, 'sound/effects/meteorimpact.ogg', 100, TRUE)
+		tearing_wall = FALSE
+		return
+	if(isliving(target)) //Swallows corpses like a snake to regain health.
+		var/mob/living/L = target
+		if(L.stat == DEAD)
+			if(is_swallowing)
+				return
+			to_chat(src, "<span class='warning'>You begin to swallow [L] whole...</span>")
+			is_swallowing = TRUE
+			if(do_after(src, 3 SECONDS, target = L))
+				is_swallowing = FALSE
+				if(eat(L))
+					adjustHealth(-L.maxHealth * 0.25)
+			return
+	. = ..()
+	if(istype(target, /obj/mecha))
+		var/obj/mecha/M = target
+		M.take_damage(50, BRUTE, "melee", 1)
+
+/mob/living/simple_animal/hostile/space_dragon/Move()
+	if(!using_special)
+		..()
```

Comment:
Will this prevent all moves like shuttle movement?

Reply:
I have no idea